Core/DistortionPointHandler.cs

```diff
@@ -0,0 +1,135 @@
+                Filters.Scene["DistortionPulse"].Load();
+            }
+        }
+
+        public override void PostUpdateProjectiles()
+        {
+            if (Main.gameMenu)
+                return;
+
+            int projectilesFound = 0;
+            int numberOfShockwaves = 0;
+            float[] progresses = new float[10];
+            float[] intensity = new float[10];
+            Vector2[] positions = new Vector2[10];
+
+            foreach (DistortionPoint point in DistortionPoints.ToArray())
+            {
+                if (!point.Active)
+                    DistortionPoints.Remove(point);
+            }
+            foreach (DistortionPoint point in DistortionPoints)
+            {
+                point.Intensity = point.UpdateIntensity.Invoke(point.Intensity, ++point.TicksPassed);
+                point.Progress = point.UpdateProgress.Invoke(point.Progress, point.TicksPassed);
+                point.Active = point.UpdateActive.Invoke(point.Progress, point.Intensity, point.TicksPassed);
+
+                intensity[projectilesFound] = point.Intensity;
+                positions[projectilesFound] = point.Position;
+                progresses[projectilesFound] = point.Progress;
+                numberOfShockwaves++;
+                projectilesFound++;
+
+                if (projectilesFound > 9)
+                    break;
+            }
+
+            if (projectilesFound == 0)
+            {
+                if (Filters.Scene["DistortionPulse"].IsActive())
+                    Filters.Scene["DistortionPulse"].Deactivate();
+
+                return;
+            }
+
+            while (projectilesFound < 9)
+            {
+                projectilesFound++;
+                progresses[projectilesFound] = 0;
+                positions[projectilesFound] = Vector2.Zero;
+                intensity[projectilesFound] = 0;
+            }
+
+            Filters.Scene["DistortionPulse"].GetShader().Shader.Parameters["progresses"].SetValue(progresses);
+            Filters.Scene["DistortionPulse"].GetShader().Shader.Parameters["positions"].SetValue(positions);
+            Filters.Scene["DistortionPulse"].GetShader().Shader.Parameters["intensity"].SetValue(intensity);
+            Filters.Scene["DistortionPulse"].GetShader().Shader.Parameters["numberOfPoints"].SetValue(numberOfShockwaves);
+
+            if (Main.netMode != NetmodeID.Server && !Filters.Scene["DistortionPulse"].IsActive())
+            {
+                Filters.Scene.Activate("DistortionPulse").GetShader().UseProgress(0f).UseColor(Color.White.ToVector3()).UseOpacity(0.0001f);
+            }
+        }
+
+        public static void AddPoint(Vector2 position, float intensity, float progress, DistortionIntensityFunction updateIntensity, DistortionProgressFunction updateProgress, DistortionActiveFunction updateActive)
+        {
+            Vector2 offScreen = new Vector2(Main.offScreenRange);
+            if (Main.drawToScreen)
```

Comment:
single line conditionals should be inlined -- use the default code cleanup profile in VS to fix errors like this!